### ISSUE TITLE
Why is it so slow?

### ISSUE THREAD

**Issue by sdegutis** [MailCore/mailcore2#467]
First I'm doing a `[MCOIMAPSearchExpression searchAll]` on the "INBOX" folder. Then I fetch just the headers and gmail-thread-id via `[session fetchMessagesByUIDOperationWithFolder:@"INBOX" requestKind:... uids:searchAllResultUids]`. Then on each of these I do a "searchGmailThreadID" for on gmail's "All Mail" folder. I've also tried a variation that omits the first "searchAll" and just uses the `MCORangeMake(1, UINT64_MAX)1 trick that's in the readme.

Each of these intermediate steps takes about 1-2 seconds! That's significantly than gmail's near-instantaneous speed. Why is it so slow? How can I speed it up?

Basically my goal is to read my inbox threads from a ObjC desktop client, with near-gmail.com speeds. Is this possible using this lib?

For reference here's [the whole file](https://github.com/sdegutis/Email/blob/master/Email/EMAppDelegate.m).
